internal/ignore/ignore.go

```diff
@@ -0,0 +1,129 @@
+package ignore
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	domainutils "github.com/supergoodsystems/supergood-go/internal/domain-utils"
+	remoteconfig "github.com/supergoodsystems/supergood-go/internal/remote-config"
+	"github.com/supergoodsystems/supergood-go/internal/shared"
+)
+
+// ShouldIgnoreRequest evaluates based off the remote config cache whether an intercepted
+// request should be forwarded to supergood
+func ShouldIgnoreRequest(req *http.Request, rc *remoteconfig.RemoteConfig) (bool, []error) {
+	var errs []error
+	domain := domainutils.GetDomainFromHost(req.Host)
+	if domain == "" {
+		return false, errs
+	}
+
+	endpoints := rc.Get(domain)
+	if len(endpoints) == 0 {
+		return false, errs
+	}
+	for _, endpoint := range endpoints {
+		if endpoint.Action != "Ignore" {
+			continue
+		}
+		testVal, err := stringifyRequestAtLocation(req, endpoint.Location)
```

Comment:
what are the cases when this throws an error?

Reply:
fails to marshal request headers or request body into json